test/unit/requestWrapper.test.js

```diff
@@ -1,6 +1,279 @@
+          null_item: null,
+          custom_file: {
+            filename: 'custom.wav',
+            data: fs.createReadStream('../blank.wav'),
+          },
+          array_item: ['a', 'b'],
+          object_item: { a: 'a', b: 'b' },
+          no_data: {
+            contentType: 'some-type',
+          },
+        },
+      },
+    };
+
+    axios.mockImplementation(requestParams => {
+      requestParams.paramsSerializer(requestParams.params);
+      return Promise.resolve('res');
+    });
+
+    sendRequest(parameters, (err, body, res) => {
+      // assert results
+      expect(axios.mock.calls[0][0].url).toEqual(
+        'https://example.ibm.com/v1/environments/environment-id/configurations/configuration-id'
+      );
+      expect(axios.mock.calls[0][0].headers).toMatchObject({
+        'Accept-Encoding': 'gzip',
+        'test-header': 'override-header-value',
+        'add-header': 'add-header-value',
+      });
+      expect(axios.mock.calls[0][0].headers['content-type']).toMatch(
+        'multipart/form-data; boundary=--------------------------'
+      );
+      expect(axios.mock.calls[0][0].httpsAgent.options.rejectUnauthorized).toEqual(true);
+      expect(axios.mock.calls[0][0].method).toEqual(parameters.defaultOptions.method);
+      expect(axios.mock.calls[0][0].params).toEqual(parameters.options.qs);
+      expect(axios.mock.calls[0][0].responseType).toEqual('json');
+      expect(res).toEqual('res');
+      expect(axios.mock.calls.length).toBe(1);
+      done();
+    });
+  });
+
+  it('should send a request with form data', done => {
+    const parameters = {
+      defaultOptions: {
+        form: { a: 'a', b: 'b' },
+        qs: {
+          version: '2017-10-15',
+        },
+        method: 'POST',
+        rejectUnauthorized: true,
+        url: 'https://example.ibm.com',
+        headers: {
+          'test-header': 'test-header-value',
+        },
+      },
+      options: {
+        url: '/v1/environments/{environment_id}/configurations/{configuration_id}',
+        path: {
+          environment_id: 'environment-id',
+          configuration_id: 'configuration-id',
+        },
+        rejectUnauthorized: false,
+        method: 'GET',
+        qs: {
+          version: '2018-10-15',
+        },
+        headers: {
+          'test-header': 'override-header-value',
+          'add-header': 'add-header-value',
+        },
+      },
+    };
+
+    axios.mockImplementation(requestParams => {
+      requestParams.paramsSerializer(requestParams.params);
+      return Promise.resolve('res');
+    });
+
+    sendRequest(parameters, (err, body, res) => {
+      // assert results
+      expect(axios.mock.calls[0][0].data).toEqual('a=a&b=b');
+      expect(axios.mock.calls[0][0].url).toEqual(
+        'https://example.ibm.com/v1/environments/environment-id/configurations/configuration-id'
+      );
+      expect(axios.mock.calls[0][0].headers).toEqual({
+        'Accept-Encoding': 'gzip',
```

Comment:
For the last test, will you add a custom value for `Accept-Encoding` and check for that? Something like `compress` is another possible option. I know we have 100% coverage but I don't see a check for that and it would be good for regressions. Let me know if you did test for that and I missed it